influxql/select.go

```diff
@@ -850,13 +857,11 @@ func (b *exprIteratorBuilder) buildCallIterator(expr *Call) (Iterator, error) {
 			}
 			return itr, nil
 		case "median":
-			opt := b.opt
-			opt.Ordered = true
-			input, err := buildExprIterator(expr.Args[0].(*VarRef), b.ic, b.sources, opt, false)
+			input, err := buildOrderedExprIterator(expr.Args[0].(*VarRef), b.ic, b.sources, b.opt, false)
 			if err != nil {
 				return nil, err
 			}
-			return newMedianIterator(input, opt)
+			return newMedianIterator(input, b.opt)
```

Comment:
Updated. It looks like the code that does this sorting is exclusively for sorting the results of selectors so it's not relevant to median or percentile, but I'll take care of modifying that in a future refactor whenever we have an actionable plan for that.